lib/mayaUsd/ufe/wrapUtils.cpp

```diff
@@ -81,36 +88,103 @@ std::string stagePath(UsdStageWeakPtr stage)
     auto segmentStrings = TfStringTokenize(ufePathString, ",");
 
-    // If there's just one segment, it's the Maya Dag path segment, so it can't
-    // have a prim.
-    if (segmentStrings.size() == 1) {
-        return UsdPrim();
+    // If there are fewer than two segments, there cannot be a USD segment, so
+    // return an invalid path.
+    if (segmentStrings.size() < 2u) {
+        return path;
     }
 
-    // We have the path string split into segments.  Build up the Ufe::Path one
-    // segment at a time.  The path segment separator is the first character
-    // of each segment.  We know that USD's separator is '/' and Maya's
+    // We have the path string split into segments. Build up the Ufe::Path one
+    // segment at a time. The path segment separator is the first character
+    // of each segment. We know that USD's separator is '/' and Maya's
     // separator is '|', so use a map to get the corresponding UFE run-time ID.
-    Ufe::Path                        path;
-    static std::map<char, Ufe::Rtid> sepToRtid = { { '/', ufe::getUsdRunTimeId() }, { '|', 1 } };
-    for (std::size_t i = 0; i < segmentStrings.size(); ++i) {
+    static std::map<char, Ufe::Rtid> sepToRtid
+        = { { '/', ufe::getUsdRunTimeId() }, { '|', ufe::getMayaRunTimeId() } };
+    for (size_t i = 0u; i < segmentStrings.size(); ++i) {
         const auto& segmentString = segmentStrings[i];
-        char        sep = segmentString[0];
+        char        sep = segmentString[0u];
         path = path + Ufe::PathSegment(segmentString, sepToRtid.at(sep), sep);
     }
+
+    return path;
+}
+#endif
+
+std::string stripInstanceIndexFromUfePath(const std::string& ufePathString)
+{
+#ifdef UFE_V2_FEATURES_AVAILABLE
+    const Ufe::Path path = Ufe::PathString::path(ufePathString);
+#else
+    const Ufe::Path path = _UfeV1StringToUsdPath(ufePathString);
+#endif
+
+    return ufe::stripInstanceIndexFromUfePath(path).string();
+}
+
+UsdPrim ufePathToPrim(const std::string& ufePathString)
+{
+#ifdef UFE_V2_FEATURES_AVAILABLE
+    return ufe::ufePathToPrim(Ufe::PathString::path(ufePathString));
+#else
+    Ufe::Path       path = _UfeV1StringToUsdPath(ufePathString);
+
+    // If there are fewer than two segments, there cannot be a USD segment, so
+    // return an invalid UsdPrim.
+    const size_t numSegments =
+#if UFE_PREVIEW_VERSION_NUM >= 2006
+        path.nbSegments();
+#else
+        path.getSegments().size();
+#endif
+    if (numSegments < 2u) {
+        return UsdPrim();
+    }
+
     return ufe::ufePathToPrim(path);
 #endif
 }
 
+int ufePathToInstanceIndex(const std::string& ufePathString)
+{
+#ifdef UFE_V2_FEATURES_AVAILABLE
+    return ufe::ufePathToInstanceIndex(Ufe::PathString::path(ufePathString));
+#else
+    Ufe::Path path = _UfeV1StringToUsdPath(ufePathString);
+
+    // If there are fewer than two segments, there cannot be a USD segment, so
+    // return ALL_INSTANCES.
+    const size_t numSegments =
+#if UFE_PREVIEW_VERSION_NUM >= 2006
```

Comment:
Same as line 152.

Reply:
Addressed in 34329b23f94983b0fb48c1104f197b99a36caf24.